ecosystem/typescript/sdk_v2/src/api/transaction_submission.ts

```diff
@@ -0,0 +1,171 @@
+import { AptosConfig } from "./aptos_config";
+import { Account } from "../core/account";
+import { AccountAuthenticator } from "../transactions/authenticator/account";
+import { MimeType, postAptosFullNode } from "../client";
+import {
+  AnyRawTransaction,
+  GenerateMultiAgentRawTransactionInput,
+  MultiAgentTransaction,
+  GenerateTransactionInput,
+  FeePayerTransaction,
+  GenerateFeePayerRawTransactionInput,
+  GenerateSingleSignerRawTransactionInput,
+  SingleSignerTransaction,
+  SimulateTransactionData,
+} from "../transactions/types";
+import { UserTransactionResponse, PendingTransactionResponse } from "../types";
+import {
+  generateSignedTransaction,
+  generateSignedTransactionForSimulation,
+  generateTransaction,
+  generateTransactionPayload,
+  signTransaction,
+} from "../transactions/transaction_builder/transaction_builder";
+
+export class TransactionSubmission {
+  readonly config: AptosConfig;
+
+  constructor(config: AptosConfig) {
+    this.config = config;
+  }
+
+  /**
+   * We are defining function signatures, each with its specific input and output.
+   * These are the possible function signature for `generateTransaction` function.
+   * When we call `generateTransaction` function with the relevant type properties,
+   * Typescript can infer the return type based on the appropriate function overload.
+   */
```

Comment:
Wonder if this shows up in the ts-docs, when we eventually use those.

Reply:
would we?